crates/eips/src/eip4844/builder.rs

```diff
@@ -192,10 +195,12 @@ impl SimpleCoder {
     /// Decode an some bytes from an iterator of valid FEs.
     ///
     /// Returns `Ok(Some(data))` if there is some data.
-    /// Returns `Ok(None)` if there is no data (length prefix is 0).
+    /// Returns `Ok(None)` if there is no data (empty iterator, length prefix is 0).
     /// Returns `Err(())` if there is an error.
     fn decode_one<'a>(mut fes: impl Iterator<Item = WholeFe<'a>>) -> Result<Option<Vec<u8>>, ()> {
-        let first = fes.next().ok_or(())?;
+        let Some(first) = fes.next() else {
+            return Ok(None);
+        };
```

Comment:
this also makes sense, because before this would always return an error for iters len > 1, right?